### PR TITLE
x#xdeploy: add the possibility of building in a container

### PR DESCRIPTION
**Pull request by pkeroulas**
This provides a consistent and controlled build environment while keeping
a native sane environment.

* a docker wrapper script is added
* the docker image init relies on the improved setup script
* the doc is refactored to help understanding the multiple build cases
* the build had a minor variable expansion issue which caused an exit

### CODE REVIEW

**Comment by pkeroulas** [comment]
Related to #15.

File: scripts/setup-dev-env.sh [hunk 1]
```diff
@@ -1,18 +1,21 @@
 #!/bin/bash
 
-apt install -y uuid-dev libpcap-dev
+# get cmake 3.9
```
[the remark this answers is not in the document]
Comment (pkeroulas) [reply scripts/setup-dev-env.sh]: Our Docker image is based on Debian Stretch for which there is package for 3.9.https://packages.debian.org/stretch-backports/cmake
Do you recommend to build Cmake from sources or something else?

[the remark this answers is not in the document]
Comment (promgamer) [reply scripts/setup-dev-env.sh]: Yes, building from source! Anyway, we should first verify if cmake exists on the system and if not, then we wget it and build it maybe :package:

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,5 @@
 /build*
-/.vs/*
-/.vscode
+/.*
 /cmake-build*
-/.idea/
 /release
+Dockerfile
```
[the remark this answers is not in the document]
Comment (pkeroulas) [reply .gitignore]: Alright. The script wraps `deploy.sh` so the Dockerfile should be located in `release/`. This folder already contains the runtime Dockerfile but this one can be moved to `release/server`

**Comment by promgamer** [comment]
Got this error:

```
mpoeira@pi-list$ ./scripts/docker_build_wrapper.sh build
docker: Error response from daemon: OCI runtime create failed: container_linux.go:348: starting container process caused "exec: \"./script/deploy/deploy.sh\": stat ./script/deploy/deploy.sh: no such file or directory": unknown.
```

**Comment by promgamer** [comment]
Just tested and it works perfectly. I think it's ready to be merged @pedro-alves-ferreira ! Well done @pkeroulas :muscle:

**Comment by pedro-alves-ferreira** [comment]
Done. Thanks @pkereloulas!

**Pull request merged**